docs_for_developer_console

Preeticp opened the pull request:
- This PR adds assemblies and modules for the Developer perspective in the OpenShift web console.
- The assemblies are part of three buckets: web console; applications/application life cycle management; applications/projects (module included as part of existing assembly).
- The content has been approved by the DevConsole SMEs and QE and will be verified once more by QE once we get a preview of this PR.
- I have received approvals for the three external links from the GSS
- The product GAs with the OCP 4.2 release.
- This PR applies to branch/enterprise-4.2

Preeticp commented:
@ahardin-rh  and @bmcelvee here's the content for Developer Perspective, it would be great if you could review it, please. 
The content has been reviewed by the ODC SMEs and the QE in gdoc format. The QE would like to take a look at the preview of this PR too, but strangely the preview link has not been generated, probably cause of the build failures in the initial commits. I have informed Vikram about the same.

ahardin-rh commented on web-console/odc-about-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,32 @@
+
+The *Developer* perspective provides the following developer focused features:
+
+* Ability to create and deploy applications on {product-title} by importing existing codebases, images, and dockerfiles.
```
The ability to (for each instance in this list)

ahardin-rh commented on web-console/odc-about-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,32 @@
+* Ability to create and deploy applications on {product-title} by importing existing codebases, images, and dockerfiles.
+* Visual representation of all the applications within a project, their build status, and the components and services associated with the applications.
+* Ability to add required applications and services from the Developer catalog and group the components of an application.
+* Integration of the Serverless capabilities (Technical Preview).
```
we say Technology Preview instead of Technical Preview per https://access.redhat.com/support/offerings/techpreview

ahardin-rh commented on applications/application-life-cycle-management/odc-creating-an-application-using-the-developer-perspective.adoc:
```diff
@@ -0,0 +1,26 @@
+
+toc::[]
+
+An application is an abstraction composed of one or more components. The *Developer* perspective in the web console provides you the following options from the *Add* view to create applications and associated services and deploy them on {product-title}:
```
I don't think we need to define what an application is here. Perhaps remove the first sentence.

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
++
+For this tutorial you do not need to use the advanced options.
+
+. In the *Builder* section, select the *Node.js*  builder image to see the details of the builder image. If required, you can change the version using the *Builder Image Version* drop-down.
```
drop-down menu.

Preeticp replied:
Used drop-down lists instead, since they are part of a form and there are no menus.

ahardin-rh commented:
@Preeticp I will continue this review over the weekend if I can, but will prioritize this for Monday.

Preeticp commented:
Sure, @ahardin-rh thank you. I am unwell myself too, will fix these by my Monday.

ahardin-rh commented on applications/application-life-cycle-management/odc-creating-an-application-using-the-developer-perspective.adoc:
```diff
@@ -0,0 +1,26 @@
+* *From Git*: Use this option to import an existing codebase in a Git repository to create, build, and deploy an application on {product-title}.
+* *Container Image*: Use existing images from an image stream or registry to deploy it on to {product-title}.
+* *From Catalog*: Explore the *Developer Catalog* to select the required applications, services, or source to image builders and add it to your project.
+* *From Dockerfile*: Import a dockerfile from your git repository to build and deploy an application.
```
Git repository

ahardin-rh commented on modules/odc-accessing-developer-perspective.adoc:
```diff
@@ -0,0 +1,30 @@
+[id="accessing-developer-perspective_{context}"]
+= Accessing Developer Perspective
+
+The *Developer* perspective in the {product-title} web console provides work flows specific to developer use cases.
```
s/work flows/workflows

ahardin-rh commented on modules/odc-accessing-developer-perspective.adoc:
```diff
@@ -0,0 +1,30 @@
+
+.Prerequisites
+
+* Ensure that you have link:https://docs.openshift.com/container-platform/latest/web-console/web-console.html[logged in to the web console].
```
for internal references, we must use this format for links:
https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#internal-cross-references

ahardin-rh commented on modules/odc-accessing-developer-perspective.adoc:
```diff
@@ -0,0 +1,30 @@
+
+.Procedure
+
+. Login to the {product-title} web console using your login credentials. The default view for the {product-title} web console is the *Administrator* perspective.
```
s/Login/Log in (for the first instance; login credentials is correct)

ahardin-rh commented on modules/odc-connecting-components-within-an-application-and-across-applications.adoc:
```diff
@@ -0,0 +1,31 @@
+
+.Procedure
+
+. Move the mouse pointer over the MongoDB service to see a dangling arrow on the node.
```
s/Move the mouse pointer over/Hover over

ahardin-rh commented on modules/odc-creating-a-project-using-the-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,25 @@
+
+.Prerequisites
+
+* Ensure that you have the appropriate roles and permissions to create projects, applications, and other workloads in {product-title}. See link:https://docs.openshift.com/container-platform/3.4/architecture/additional_concepts/authorization.html#roles[here] for more details on roles and permissions.
```
We should avoid "here" statements when linking; maybe just have:
xref:../architecture/additional_concepts/authorization.adoc#roles[Learn more about roles and permissions].

Also, note the link format of an xref with a .adoc instead of .html

ahardin-rh commented on modules/odc-creating-a-project-using-the-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,25 @@
+* Ensure that you have the appropriate roles and permissions to create projects, applications, and other workloads in {product-title}. See link:https://docs.openshift.com/container-platform/3.4/architecture/additional_concepts/authorization.html#roles[here] for more details on roles and permissions.
+
+.Procedure
+You can create a project using the *Developer* perspective, as follows:
```
We embolden Developer here, but not in other places. We should check all instances and keep consistent with usage.

ahardin-rh commented on modules/odc-creating-a-project-using-the-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,25 @@
+.Procedure
+You can create a project using the *Developer* perspective, as follows:
+
+. In the *Add* view, click the *Project* drop-down to see a list of all available projects. Select *Create Project*.
```
drop-down menu

ahardin-rh commented on modules/odc-creating-a-project-using-the-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,25 @@
++
+image::odc_create_project.png[Create Project]
+
+. In the *Create Project* dialog box, enter a unique name for the *Name* field. For this tutorial, enter `myproject` as the name of the project in the *Name* field.
```
I avoid referring to our docs as individual tutorials, or anything similar. Perhaps rewrite this as:

    For example, enter `myproject` as the name of the project in the *Name* field.

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+= Grouping multiple components within an application
+
+You can use the *Add* page to add multiple components or services to your project and use the *Topology* page to group applications and resources within an application group.
+For this tutorial, add a MongoDB database service to an existing application with a Node.js component as follows:
```
Consider rewriting as:
The following procedure adds a MongoDB database service to an existing application with a Node.js component.

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+.Procedure
+
+. Create and deploy the MongoDB service to the `myproject` project as follows:
+.. In the Developer Perspective, navigate to the *+Add* view and select the *Database* option to see the *Developer Catalog*, which has multiple options that you can add as components or services to your application.
```
*Developer* perspective  (to be consistent with other sections)

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+. Create and deploy the MongoDB service to the `myproject` project as follows:
+.. In the Developer Perspective, navigate to the *+Add* view and select the *Database* option to see the *Developer Catalog*, which has multiple options that you can add as components or services to your application.
+.. Click on the *MongoDB* option to see the details for the service.
+.. Click *Instantiate Template* to see an auto populated template with details for the MongoDb service and click *Create* to create the service.
```
s/auto populated/automatically populated
MongoDB service, and click

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+.. Click *Instantiate Template* to see an auto populated template with details for the MongoDb service and click *Create* to create the service.
+
+. On the left navigation panel, click *Topology* to see the MongoDB service deployed in your project.
+. To add the MongoDB service to the existing application group, select the *mongodb* Pod and drag it to the application, the MongoDB service is added to the existing application group.
```
application; the

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+
+Alternatively, you can also add the component to an application as follows:
+
+. To add the MongoDB service to your application, click on the *mongodb* Pod to see the *Overview* panel to the right.
```
s/upper-right/upper right

ahardin-rh commented on modules/odc-grouping-multiple-components-within-an-application.adoc:
```diff
@@ -0,0 +1,35 @@
+Alternatively, you can also add the component to an application as follows:
+
+. To add the MongoDB service to your application, click on the *mongodb* Pod to see the *Overview* panel to the right.
+. Click the *Actions* drop-down on the upper-right of the panel and select *Edit Application Grouping*.
```
click the *Select an Application* drop-down menu, and select 

(drop-down is an adjective according to IBM style)

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+.Prerequisites
+
+* Ensure that you have link:https://docs.openshift.com/container-platform/latest/web-console/web-console.html[logged in to the web console].
+* Ensure that you have switched to the *Developer* perspective.
```
Ensure that you are in the *Developer* perspective.

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+
+.Prerequisites
+
+* Ensure that you have link:https://docs.openshift.com/container-platform/latest/web-console/web-console.html[logged in to the web console].
```
Edit the link per our guidelines https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#internal-cross-references

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+. Optionally, you can click *Show Advanced Git Options*  to add details such as:
+
+* *Git Reference* to point to code in a specific branch, tag, or commit to be used to build the application.
+* *Context Dir* to specify the subdirectory for the application source code, you want to use to build the application.
```
source code you want

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+. The *Serverless* section is displayed in the *Import from git* form only if the link:https://docs.openshift.com/container-platform/4.1/serverless/installing-openshift-serverless.html#installing-serverless-operator_installing-openshift-serverless[*Serverless Operator*] is installed in your cluster.
+Select *Enable scaling to zero when idle* to create a serverless application that auto scales your Pods to zero to prevent consumption of resources when idle.
++
+NOTE: The *Serverless* option is a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature.
```
Notes should be formatted like this: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#admonitions

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+NOTE: The resource name must be unique in a namespace. Modify the resource name if you get an error.
+
+. The *Serverless* section is displayed in the *Import from git* form only if the link:https://docs.openshift.com/container-platform/4.1/serverless/installing-openshift-serverless.html#installing-serverless-operator_installing-openshift-serverless[*Serverless Operator*] is installed in your cluster.
+Select *Enable scaling to zero when idle* to create a serverless application that auto scales your Pods to zero to prevent consumption of resources when idle.
```
automatically scales

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
++
+NOTE: The resource name must be unique in a namespace. Modify the resource name if you get an error.
+
+. The *Serverless* section is displayed in the *Import from git* form only if the link:https://docs.openshift.com/container-platform/4.1/serverless/installing-openshift-serverless.html#installing-serverless-operator_installing-openshift-serverless[*Serverless Operator*] is installed in your cluster.
```
Change link to an xref

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+
+Routing::
+Click the *Routing* link to:
+* Customize the hostname for the route.
```
s/hostname/host name

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+For serverless applications, the Knative Service manages all the routing options above. However, you can customize the target port for traffic, if required. If the target port is not specified, the default port of `8080` is used.
+
+Build and Deployment Configuration::
+Click the *Build Configuration* and *Deployment Configuration* links to see the respective configuration options. Some of the options are selected by default, you can customize them further by adding the necessary triggers and environment variables. For serverless applications, the *Deployment Configuration* option is not displayed as the Knative configuration resource maintains the desired state ...
```
default;  you can

ahardin-rh commented on modules/odc-importing-a-codebase-from-git-to-create-an-application.adoc:
```diff
@@ -0,0 +1,76 @@
+For serverless applications, you can:
+
+* Set the upper and lower limit for the number of pods that can be set by the autoscaler. If the lower limit is not specified, it defaults to zero.
+* Define the soft limit for the required number of concurrent requests per instance of the application at a given time. It is the recommended configuration for autoscaling. If not specified, it takes the value specified in the cluster config.
```
s/config/configuration

ahardin-rh commented on modules/odc-interacting-with-the-application.adoc:
```diff
@@ -0,0 +1,20 @@
+[id="odc-interacting-with-the-application_{context}"]
+= Interacting with the application
+
+The Topology view in the Developer perspective of the web console provides the following options to interact with the application:
```
*Developer* perspective

ahardin-rh replied:
Is it really a procedure if it's just outlining options?

ahardin-rh commented on modules/odc-interacting-with-the-application.adoc:
```diff
@@ -0,0 +1,20 @@
++
+If the *Eclipse Che* Operator is installed in your cluster, a Che workspace (image:odc_che_workspace.png[title="Che Workspace"]) is created and you are directed to the workspace to edit your source code. If it is not installed, you will be directed to the Git repository (image:odc_git_repository.png[title="Git Repository"]) your source code is hosted in.
+
+. Move your cursor over the lower left icon on the Pod to see the name of the latest build and its status. The status of the application build is indicated as New (image:odc_build_new.png[title="New Build"]), Pending (image:odc_build_pending.png[title="Pending Build"]), Running (image:odc_build_running.png[title="Running Build"]), Completed (image:odc_build_completed.png[title="Completed Build"]) ...
```
s/move/hover

ahardin-rh replied:
I would embolden each status type. You do so in other areas.

ahardin-rh commented on modules/odc-scaling-application-pods-and-checking-builds-and-routes.adoc:
```diff
@@ -0,0 +1,24 @@
+[id="odc-scaling-application-pods-and-checking-builds-and-routes_{context}"]
+= Scaling application pods and checking builds and routes
+
+The topology view provides the details of the deployed components in the *Overview* panel. You can use the *Overview* and *Resources* tabs to scale the application Pods, check build status, services, and routes as follows:
```
Topology view

ahardin-rh replied:
Perhaps embolden as well since you do that in the next file?

ahardin-rh commented on modules/odc-scaling-application-pods-and-checking-builds-and-routes.adoc:
```diff
@@ -0,0 +1,24 @@
+The topology view provides the details of the deployed components in the *Overview* panel. You can use the *Overview* and *Resources* tabs to scale the application Pods, check build status, services, and routes as follows:
+
+.Procedure
+
```
I think this is more reference than procedure

ahardin-rh commented on modules/odc-scaling-application-pods-and-checking-builds-and-routes.adoc:
```diff
@@ -0,0 +1,24 @@
+
+. Click on the component node to see the *Overview* panel to the right. Use the *Overview* tab to:
+
+* Scale your Pods using the up and down arrows to increase or decrease the number of instances of the application manually. For serverless applications, the Pods are auto scaled down to zero when idle and scaled up depending on the channel traffic.
```
automatically scaled

ahardin-rh commented on web-console/odc-about-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,32 @@
+ifdef::context[:parent-context-of-odc-about-developer-perspective-in-the-web-console: {context}]
+[id="odc-about-developer-perspective-in-the-web-console_{context}"]
+= About Developer perspective in the web console
```
the Developer perspective

ahardin-rh commented on web-console/odc-about-developer-perspective-in-the-web-console.adoc:
```diff
@@ -0,0 +1,32 @@
+
+The {product-title} web console provides two perspectives; the *Administrator* perspective and the *Developer* perspective.
+
+The *Developer* perspective provides the following developer focused features:
```
developer-focused

ahardin-rh replied:
The *Developer* perspective provides the ability to:
    
    * Create and deploy ...
    * Visually interact ... (and so on)

ahardin-rh commented:
@Preeticp This is looking great. Some miscellaneous minor comments from me :+1:

vikram-redhat commented on modules/applications-create-using-cli-template.adoc:
```diff
@@ -1,6 +1,6 @@
-// * applications/creating-new-applications.adoc
+// * applications/application-life-cycle-management/creating-applications-using-the-cli.adoc
 
 [id="applications-create-using-cli-template-{context}"]
```
@Preeticp:  _{context}

vikram-redhat commented on _topic_map.yml:
```diff
@@ -168,8 +168,11 @@ Topics:
 - Name: Configuring the web console
   File: configuring-web-console
+- Name: Developer perspective
+  File: odc-about-developer-perspective-in-the-web-console
```
@Preeticp  I don't think we should use 'in', 'the' as part of file names, contexts or anchors.

vikram-redhat commented on _topic_map.yml:
```diff
@@ -476,8 +479,12 @@ Topics:
+  - Name: Creating applications using the CLI
+    File: creating-applications-using-the-cli
+  - Name: Creating an application using the Developer perspective
+    File: odc-creating-an-application-using-the-developer-perspective
```
@Preeticp same here, please remove all instances of 'the', 'an' etc.

vikram-redhat commented on applications/application-life-cycle-management/odc-creating-an-application-using-the-developer-perspective.adoc:
```diff
@@ -0,0 +1,26 @@
+ifdef::context[:parent-context-of-odc-creating-an-application-using-the-developer-perspective: {context}]
```
@Preeticp we don't use parent contexts.

vikram-redhat commented on applications/application-life-cycle-management/odc-viewing-application-composition-using-the-topology-view.adoc:
```diff
@@ -0,0 +1,34 @@
+ifdef::context[:parent-context-of-odc-viewing-application-composition-using-the-topology-view: {context}]
```
Same here. No parent contexts.

Preeticp commented:
Please do not merge yet, I want to update a few screenshots and links tomorrow.

kalexand-rh commented on modules/odc-accessing-developer-perspective.adoc:
```diff
@@ -0,0 +1,30 @@
+
+.Prerequisites
+
+* Ensure that you have xref:../web-console/web-console.adoc[logged in to the web console].
```
remove this xref.

kalexand-rh commented on modules/odc-creating-projects-using-developer-perspective.adoc:
```diff
@@ -0,0 +1,25 @@
+
+.Prerequisites
+
+* Ensure that you have the appropriate xref:../authentication/using-rbac.adoc#default-roles_using-rbac[roles and permissions] to create projects, applications, and other workloads in {product-title}.
```
remove this xref

kalexand-rh commented on modules/odc-importing-codebase-from-git-to-create-application.adoc:
```diff
@@ -0,0 +1,82 @@
+
+.Prerequisites
+
+* Ensure that you have xref:../web-console/web-console.adoc[logged in to the web console].
```
remove this xref

kalexand-rh commented on modules/odc-importing-codebase-from-git-to-create-application.adoc:
```diff
@@ -0,0 +1,82 @@
++
+[NOTE]
+====
+* The *Serverless* section is displayed in the *Import from git* form only if the xref:../serverless/installing-openshift-serverless.adoc#installing-serverless-operator_installing-openshift-serverless[*Serverless Operator*] is installed in your cluster.
```
remove this xref

Preeticp commented:
@ahardin-rh , @vikram-redhat @kalexand-rh thank you so much for all your inputs. I have fixed all the review comments. Finally the build is all fine :) and I have squashed all my commits. Let me know if I need to fix anything else.

ahardin-rh commented:
Closing in favor of https://github.com/openshift/openshift-docs/pull/16935 so that we can fix the conflicts and get this merged so that Preeti can continue iterating on the content in a separate PR.
